plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConversionContext.kt

```diff
@@ -10,13 +10,18 @@ import org.jetbrains.kotlin.fir.symbols.impl.FirFunctionSymbol
 import org.jetbrains.kotlin.fir.types.ConeKotlinType
 import org.jetbrains.kotlin.fir.types.resolvedType
 import org.jetbrains.kotlin.formver.PluginConfiguration
+import org.jetbrains.kotlin.formver.embeddings.FieldEmbedding
 import org.jetbrains.kotlin.formver.embeddings.MethodEmbedding
 import org.jetbrains.kotlin.formver.embeddings.TypeEmbedding
 import org.jetbrains.kotlin.formver.embeddings.VariableEmbedding
+import org.jetbrains.kotlin.formver.viper.MangledName
+import org.jetbrains.kotlin.formver.viper.ast.Field
 
 interface ProgramConversionContext {
     val config: PluginConfiguration
 
+    val fields: MutableMap<MangledName, FieldEmbedding>
```

Comment:
I would sooner expose this as an `embedField` function, by the way, it's a bit simpler an interface.

Reply:
The problem here is getting the type correct for generic fields. if you have `val box = Box<Int>(3)` and then call `box.t`, the property access expression will give you type `Int` even though the underlying field is of type `Any?`. A proper way to do it is I guess introduce as `ClassEmbedding` and store the fields of that class in it.

Reply:
Ah, I don't mean removing the map entirely; there probably still has to be one, but I'd leave it as an implementation detail of `ProgramConverter`, and only expose `embedField` to the users.

Reply:
Oh yeah that makes total sense

Reply:
I called the method `getField` instead though as it doesn't automatically embed the field if it doesn't exist but rather returns null in that case.